scipy/sparse/_sputils.py

```diff
@@ -369,14 +369,14 @@ def isintlike(x) -> bool:
     return True
 
 
-def isshape(x, nonneg=False, *, allow_nd=(2,)) -> bool:
+def isshape(x, nonneg=False, *, allow_nd=(2,), check_nd=True) -> bool:
```

Comment:
IIUC, setting `check_nd=True` is the same as if `allow_nd` were the set of all non-negative integers? Did you consider passing a special value to `allow_nd` instead of adding a new arg?
Now that I understand that `check_nd` means "check the number of dimensions" rather than "perform the check for n-d arrays", I don't care as much, but initially I was concerned about the name of the new argument. (Now I see it is consistent with `allow_nd`, and is exactly as well-named as that.)